### ISSUE TITLE
Uncaught PHP Exception Exception: "Serialization of 'Closure' is not allowed"

### ISSUE THREAD

**Issue by MNDMV** [ChristopherDosin/Shopware-6-Blog-Plugin#159]
With the update to the latest plugin version, out of nowhere we now get a generic error message for all blog pages instead of the blog page. Practically no blog entry can be displayed anymore.

Unfortunately only the following error appears in the logs:
request.CRITICAL: Uncaught PHP Exception Exception: "Serialization of 'Closure' is not allowed" at vendor/shopware/core/Framework/Adapter/Cache/CacheValueCompressor.php line 21 {"exception":"[object] (Exception(code: 0): Serialization of 'Closure' is not allowed at vendor/shopware/core/Framework/Adapter/Cache/CacheValueCompressor.php:21)"} []

For the error itself, you can quickly find this:
https://stackoverflow.com/questions/13734224/exception-serialization-of-closure-is-not-allowed

After a while I noticed that this plugin introduced 3 new classes, each handling caching:
src/custom/plugins/SasBlogModule/src/Controller/CachedBlogController.php
src/custom/plugins/SasBlogModule/src/Controller/CachedBlogRssController.php
src/custom/plugins/SasBlogModule/src/Controller/CachedBlogSearchController.php

So this is very likely where the problem is hidden. What exactly is going wrong is not known, the 3 classes do not use a "Closure" directly but they call CacheValueCompressor.

Our Shopware version is 6.4.11.1 and the plugin is now version 1.5.14.

**Comment by MNDMV** [comment]
Unfortunatlly, this is not fixed. It is still broken and as i had a little bit of time, i looked a little inside your caching approche:
you are trying to serialize a complete final response-object. For my understanding, this can't work as you never know what will be inside that. I'm sorry. The closures inside the functions are not the problem, your logic is.

**Comment by huytdq94** [comment]
Hi @MNDMV, thanks for your opinion. It would be nicer if you can give me more detail about the issue, and how to reproduce it, then we can debug the root cause and fix it better. 🙏

**Comment by MNDMV** [comment]
@huytdq94 I'm very sorry, but my time budget has unfortunately run out here.

So far my understanding in the short time I could look at it: Shopware's caching logic differs from the plugin's. While Shopware only serializes/caches certain objects that are created specifically for this purpose, the plugin tries to serialize the entire response object, with all its other objects written into it by Shopware as well as by other plugins.
I checked the pull request manually. There is also a Closure missing that is left after the pullrequest. But even then it doesn't work unfortunately. The error is not triggered because there are closures in the plugin here, but because there are some in the (response-)object to be serialized. I had to remove the new caching class from the plugin to get the plugin running again. It is of course likely that the problem is triggered by another plugin that uses closures, but caching the entire response object is probably not intended by Shopware.
